synapse/tests/test_lib_aha.py

```diff
@@ -294,3 +297,41 @@ async def test_lib_aha_loadenv(self):
                 self.len(1, s_telepath.aha_clients)
                 self.nn(info.get('client'))
                 await fini()
+
+    async def test_lib_aha_finid_cell(self):
+
+        with self.getTestDir() as dirn:
+            async with await self.aha_ctor(dirn) as aha:
+
+                cryo0_dirn = s_common.gendir(aha.dirn, 'cryo0')
+
+                host, port = await aha.dmon.listen('tcp://127.0.0.1:0')
+                await aha.auth.rootuser.setPasswd('hehehaha')
+
+                wait00 = aha.waiter(1, 'aha:svcadd')
+                conf = {
+                    'aha:name': '0.cryo.mynet',
+                    'aha:admin': 'root@cryo.mynet',
+                    'aha:registry': [f'tcp://root:hehehaha@127.0.0.1:{port}',
+                                     f'tcp://root:hehehaha@127.0.0.1:{port}'],
+                    'dmon:listen': 'tcp://0.0.0.0:0/',
+                }
+                async with self.getTestCryo(dirn=cryo0_dirn, conf=conf) as cryo:
+
+                    await cryo.auth.rootuser.setPasswd('secret')
+
+                    ahaadmin = await cryo.auth.getUserByName('root@cryo.mynet')
+                    self.nn(ahaadmin)
+                    self.true(ahaadmin.isAdmin())
+
+                    await wait00.wait(timeout=2)
+
+                    async with await s_telepath.openurl('aha://root:secret@0.cryo.mynet') as proxy:
+                        self.nn(await proxy.getCellIden())
+
+                    await aha.fini()
+
+                    with self.raises(s_exc.NoSuchObj):
+
+                        async with await s_telepath.openurl('aha://root:secret@0.cryo.mynet') as proxy:
```

Comment:
Lets talk about this fail case, it feels a bit weird to raise NoSuchObj from telepath.openurl() during a lookup via an optional resolver.